Week 4/gemCollector/src/GemCollector.java

```diff
@@ -124,15 +131,82 @@ public void pushNextPoint(int y, int x, int collectedGem, boolean[][] pathMap) {
     Point nextPoint = new Point(y, x);
     int nextCollectedGem = collectedGem;
     boolean[][] nextPathMap = new boolean[mapSizeY][mapSizeX];
-    // nextPathMap = pathMap;
     deepCopyArray(pathMap, nextPathMap);
     PointWithMap nextPointWithMap = new PointWithMap(nextPoint, nextPathMap, nextCollectedGem);
 
     dfsStack.push(nextPointWithMap);
   }
 
-  public boolean[][] getAnswerPath() {
-    return answerPath;
+  // Using the given mapSizeY and mapSizeX, initialize gem map, obstacle map and answer path.
+  // All points in maps should be initialized to false.
+  public void initializeMaps() {
+    gemMap = new boolean[mapSizeY][mapSizeX];
+    obstacleMap = new boolean[mapSizeY][mapSizeX];
+    answerPath = new boolean[mapSizeY][mapSizeX];
+
+    for (int i = 0; i < mapSizeY; ++i) {
+      for (int j = 0; j < mapSizeX; ++j) {
+        gemMap[i][j] = false;
+        obstacleMap[i][j] = false;
+        answerPath[i][j] = false;
+      }
+    }
+  }
+
+  public void setStratPoint(int y, int x) {
+    startPoint.setY(y);
+    startPoint.setX(x);
+  }
+
+  public void setEndPoint(int y, int x) {
+    endPoint.setY(y);
+    endPoint.setX(x);
+  }
+
+  public int getMapSizeX() {
+    return mapSizeX;
+  }
+
+  public void setMapSizeX(int mapSizeX) throws GemMapInputException {
+    if (mapSizeX > 15) {
+      throw new GemMapInputException("Map size should not exceed 15x15.");
+    }
+    this.mapSizeX = mapSizeX;
+  }
+
+  public int getMapSizeY() {
+    return mapSizeY;
+  }
+
+  public void setMapSizeY(int mapSizeY) throws GemMapInputException{
+    if (mapSizeY > 15) {
+      throw new GemMapInputException("Map size should not exceed 15x15.");
+    }
+    this.mapSizeY = mapSizeY;
+  }
+
+  public void setGemMap(int y, int x, boolean flag) {
+    gemMap[y][x] = flag;
+  }
+
+  public void setObstacleMap(int y, int x, boolean flag) {
+    obstacleMap[y][x] = flag;
+  }
```

Comment:
I think you should not separate gem map and obstacle map
Your separate makes too many redundant codes